### PR TITLE
Documentation - topics/search section migration to markdown

### PR DESCRIPTION
**Pull request by lb-**
-  Part of https://github.com/wagtail/wagtail/issues/8383
-  Note: a few small changes have been made
  - Less reliance on multiple `note` admonitions in a row
  - Remove `tip` usage and replace with `note` admonition as per recommendations in documentation writing guide.
  - Reduce heading nesting in searching.md - it appeared to be heavily nested in a way that did not flow well, the 'changing search behaviour' is now a h2 instead of h3 with the headings under that moving up one level.
-   [x] Does the code comply with the style guide?
-   [x] For new features: Has the documentation been updated accordingly?
-   **Important:** Intentionally two commits to preserve the history of the file with git.

### This also fixes a bug with a link shown in screenshot below.

| Before      | After |
| ----------- | ----------- |
| Screen Shot 2022-06-25 at 10 49 40 pm  |  Screen Shot 2022-06-25 at 11 29 24 pm  |

### Section heading comparison
Comparison of the way the sections are nested now.

| Before      | After |
| ----------- | ----------- |
|   Screen Shot 2022-06-25 at 11 30 26 pm    |    Screen Shot 2022-06-25 at 11 30 51 pm    |

### CODE REVIEW

File: docs/topics/search/backends.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+### Database Backend (default)
+
+`wagtail.search.backends.database`
+
+The database search backend searches content in the database using the full text search features of the database backend in use (such as PostgreSQL FTS, SQLite FTS5).
+This backend is intended to be used for development and also should be good enough to use in production on sites that don't require any Elasticsearch specific features.
+
+```{versionchanged} 2.15
+``wagtail.search.backends.database`` replaces the old ``wagtail.search.backends.db`` backend which works using simple substring matching only. ``wagtail.search.backends.db`` is still the default if ``WAGTAILSEARCH_BACKENDS`` is not specified; ``wagtail.search.backends.database`` will become the default in Wagtail 2.17.
```
Comment (zerolab) [inline docs/topics/search/backends.md:62]: question should the inline code samples be in double backticks? or is this a side effect of MyST?

Comment (lb-) [inline docs/topics/search/backends.md:62]:
```suggestion
`wagtail.search.backends.database` replaces the old `wagtail.search.backends.db` backend which works using simple substring matching only. `wagtail.search.backends.db` is still the default if `WAGTAILSEARCH_BACKENDS` is not specified; `wagtail.search.backends.database` will become the default in Wagtail 2.17.
```

File: docs/topics/search/backends.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                    }
+                }
+            }
+        }
+    }
+```
+
+If you prefer not to run an Elasticsearch server in development or production, there are many hosted services available, including [Bonsai](https://bonsai.io/), who offer a free account suitable for testing and development. To use Bonsai:
+
+-   Sign up for an account at `Bonsai`\_
```
Comment (lb-) [inline docs/topics/search/backends.md:147]:
```suggestion
-   Sign up for an account at `Bonsai`
```

File: docs/topics/search/searching.md [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+All other methods of `PageQuerySet` can be used with `search()`. For example:
+
+```python
+# Search all live EventPages that are under the events index
+>>> EventPage.objects.live().descendant_of(events_index).search("Event")
+[<EventPage: Event 1>, <EventPage: Event 2>]
+```
+
+```{note}
+The ``search()`` method will convert your ``QuerySet`` into an instance of one of Wagtail's ``SearchResults`` classes (depending on backend). This means that you must perform filtering before calling ``search()``.
```
Comment (lb-) [inline docs/topics/search/searching.md:30]:
```suggestion
The `search()` method will convert your `QuerySet` into an instance of one of Wagtail's `SearchResults` classes (depending on backend). This means that you must perform filtering before calling `search()`.
```

File: docs/topics/search/searching.md [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+
+Wagtail provides a separate method which performs partial matching on specific autocomplete fields. This is useful for suggesting pages to the user in real-time as they type their query.
+
+```python
+>>> EventPage.objects.live().autocomplete("Eve")
+[<EventPage: Event 1>, <EventPage: Event 2>]
+```
+
+```{note}
+This method should only be used for real-time autocomplete and actual search requests should always use the ``search()`` method.
```
Comment (lb-) [inline docs/topics/search/searching.md:46]:
```suggestion
This method should only be used for real-time autocomplete and actual search requests should always use the `search()` method.
```

File: docs/topics/search/backends.md (hunk 1 above)
Comment (lb-) [reply docs/topics/search/backends.md]: nice pickup. Thanks. Fixed now.

**Review by zerolab — APPROVED** [review]
LGTM!

**Comment by lb-** [comment]
Thanks @zerolab

**Pull request merged**